File: 4000~4999/4673.py
def d(n):
    sum = n
    for i in str(n):
        sum += int(i)
    return sum


arr = list()
for i in range(1, 10000):
    arr.append(d(i))

for i in range(1, 10001):
    if i not in arr:
        print(i)

File: c/4000~4999/4673.c
#include <stdio.h>

int d(int n)
{
	int sum = n;
	while (n)
	{
		sum += n % 10;
		n /= 10;
	}
	return sum;
}

int main()
{
	int arr[10001] = {0};
	for (int i = 1; i < 10001; i++)
	{
		int t = d(i);
		if (t < 10001) arr[t] = 1;
	}
	for (int i = 1; i < 10001; i++)
	{
		if (!arr[i] ) printf("%d\n", i);
	}
	return 0;
}
